src/lib/y2partitioner/widgets/btrfs_table.rb

```diff
@@ -0,0 +1,164 @@
+require "yast"
+require "cwm/table"
+require "y2partitioner/widgets/blk_devices_table"
+require "y2partitioner/widgets/help"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Widgets
+    # Table widget to represent the Btrfs filesystems
+    class BtrfsTable < CWM::Table
+      include BlkDevicesTable
+      include Help
+
+      attr_reader :filesystems
+
+      # @param filesystems [Array<Y2Storage::Filesystems::BlkFilesystem>] btrfs filesystems
+      def initialize(filesystems)
+        textdomain "storage"
+        @filesystems = filesystems
+      end
+
+      # @see CWM::Table#header
+      def header
+        columns.map { |c| send("#{c}_title") }
+      end
+
+      # @see CWM::Table#items
+      def items
+        filesystems.map { |f| values_for(f) }
+      end
+
+      # Returns the filesystem of the selected row
+      #
+      # @return [Y2Storage::Filesystems::BlkFilesystem] nil if there is no selected row
+      def selected_filesystem
+        device = selected_device
+        return nil if device.nil?
+        device.filesystem
+      end
+
+      # Table help
+      #
+      # Shows help of all columns
+      def help
+        header = _(
+          "<p>This view shows all Btrfs filesystems.</p>" \
+          "<p>The table contains:</p>" \
+        )
+        fields = HELP_FIELDS.map { |f| helptext_for(f) }.join("\n")
```

Comment:
Do you test it? As help use richtext, maybe it is better to join it with `<br>` instead of `\n`.